pkgs/development/python-modules/signedjson/default.nix

```diff
@@ -1,24 +1,25 @@
 { lib
 , buildPythonPackage
-, fetchFromGitHub
+, fetchPypi
 , canonicaljson
 , unpaddedbase64
 , pynacl
 , typing-extensions
+, setuptools-scm
+, importlib-metadata
 }:
 
 buildPythonPackage rec {
   pname = "signedjson";
   version = "1.1.1";
 
-  src = fetchFromGitHub {
-    owner = "matrix-org";
-    repo = "python-${pname}";
-    rev = "v${version}";
-    sha256 = "0y5c9v4vx9hqpnca892gc9b4xgs4gp5xk6l1wma5ciz8zswp9yfs";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0280f8zyycsmd7iy65bs438flm7m8ffs1kcxfbvhi8hbazkqc19m";
   };
 
-  propagatedBuildInputs = [ canonicaljson unpaddedbase64 pynacl typing-extensions ];
+  nativeBuildInputs = [ setuptools-scm ];
+  propagatedBuildInputs = [ canonicaljson unpaddedbase64 pynacl typing-extensions importlib-metadata ];
```

Comment:
`typing-extensions` would not be needed for Python 3.7 and above.

`++ lib.optionals (pythonOlder "3.7") [ typing-extensions ]:`

`importlib.metadata` was [introduced](https://docs.python.org/3/library/importlib.metadata.html) with Python 3.8

`  ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ];`

Reply:
Apparently `typing-extensions` is needed:
```
ERROR: Could not find a version that satisfies the requirement typing-extensions>=3.5 (from signedjson)
ERROR: No matching distribution found for typing-extensions>=3.5
builder for '/nix/store/aj8lsyi88f9xw2wnmslvh1gmf92c9qbw-python3.9-signedjson-1.1.1.drv' failed with exit code 1
```
```
ERROR: Could not find a version that satisfies the requirement importlib-metadata (from signedjson)
ERROR: No matching distribution found for importlib-metadata
builder for '/nix/store/mhy3va7r5yig7a7rn8bl3i2jbvngibcm-python3.8-signedjson-1.1.1.drv' failed with exit code 1
```

I can't test python 3.10, because canonicaljson fails.
```
builder for '/nix/store/5j2liar6pj5nb0s8iy4i7bq4g5cy0478-python3.10-canonicaljson-1.4.0.drv' failed with exit code 1
```

Reply:
#120737

Reply:
yea, backport packages in python are misused, and sometimes subject to change. For example, `importlib-metadata` now exports behavior in 3.10, so it's now relevant to python3.8 and python3.9 again. https://pypi.org/project/importlib-metadata/

Reply:
@jonringer So… do I need to do something here? I don't really know much about python things, but `matrix-synapse` doesn't build because of this.

Reply:
at this point, I would just follow what upstream does, if it's broken (which is probably the case with `typing-extensions`) then it's on upstream.

Reply:
Ok. So I'll leave `typing-extensions` as is. As for `importlib-metadata`, I pulled in [this patch](https://github.com/matrix-org/python-signedjson/commit/c40c83f844fee3c1c7b0c5d1508f87052334b4e5) and made it conditional to < 3.8, as @fabaff suggested.